### PR TITLE
[CodeCamp #1535] Add SAD

### PR DESCRIPTION
**Pull request by xuan07472**
## Motivation

Add MattingSAD of MMEditing to MMEval and complete the adaptation.

## Modification

Add MattingSAD of MMEditing to MMEval
The PR link using mmeval.SAD in mmediting：https://github.com/open-mmlab/mmediting/pull/1592#issue-1525897921

### CODE REVIEW

File: mmeval/metrics/matting_sad.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        Args:
+            pred_alpha(Sequence[np.ndarray]): Pred_alpha data of predictions.
+            ori_alpha(Sequence[np.ndarray]): Ori_alpha data of data_batch.
+        """
+
+        for pred_alpha, gt_alpha in zip(
+                pred_alphas,
+                gt_alphas,
+        ):
```
Comment (ice-tong) [inline mmeval/metrics/matting_sad.py:58]:
```suggestion
        for pred_alpha, gt_alpha in zip(pred_alphas, gt_alphas):
```

File: mmeval/metrics/matting_sad.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        >>> import numpy as np
+        >>>
+        >>> mattingmse = MattingMSE()
+        >>> pred_alpha = np.zeros((32, 32), dtype=np.uint8)
+        >>> gt_alpha = np.ones((32, 32), dtype=np.uint8) * 255
+        >>> trimap = np.zeros((32, 32), dtype=np.uint8)
+        >>> trimap[:16, :16] = 128
+        >>> trimap[16:, 16:] = 255
+        >>> mattingmse(pred_alpha, gt_alpha, trimap)  # doctest: +ELLIPSIS
+        {'MattingMSE': ...}
```
Comment (ice-tong) [inline mmeval/metrics/matting_sad.py:40]: Should be update~

File: mmeval/metrics/matting_sad.py (hunk 1 above)
Comment (xuan07472) [reply mmeval/metrics/matting_sad.py:58]: Updated

File: mmeval/metrics/matting_sad.py (hunk 2 above)
Comment (xuan07472) [reply mmeval/metrics/matting_sad.py:40]: Updated

File: mmeval/metrics/matting_sad.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            Default to 1000.
+
+    Note:
+        The current implementation assumes the image / alpha / trimap
+        a numpy array with pixel values ranging from 0 to 255.
+
+        The pred_alpha should be masked by trimap before passing
+        into this metric.
+
+
```
Comment (ice-tong) [inline mmeval/metrics/matting_sad.py:27]: We only need one blank line~
```suggestion

```

File: mmeval/metrics/matting_sad.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        >>> mattingsad = MattingSAD()
+        >>> pred_alpha = np.zeros((32, 32), dtype=np.uint8)
+        >>> gt_alpha = np.ones((32, 32), dtype=np.uint8) * 255
+        >>> mattingsad(pred_alpha, gt_alpha)  # doctest: +ELLIPSIS
+        {'MattingSAD': ...}
+    """
+
+    def __init__(self, norm_const=1000, **kwargs) -> None:
+        self.norm_const = norm_const
+        super().__init__(**kwargs)
```
Comment (ice-tong) [inline mmeval/metrics/matting_sad.py:40]:
```suggestion
        super().__init__(**kwargs)
        self.norm_const = norm_const
```

File: mmeval/metrics/sad.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+from typing import Dict, List, Sequence
+
+from mmeval.core import BaseMetric
+
+
+class SAD(BaseMetric):
+    """Sum of Absolute Differences metric for image.
+
+    This metric compute per-pixel absolute difference and sum across all
+    pixels.
```
Comment (ice-tong) [inline mmeval/metrics/sad.py:12]:
```suggestion
    This metric computes per-pixel absolute difference and sum across all
    pixels.
```

**Review by ice-tong — APPROVED** [review]
LGTM with 2 comments

File: mmeval/metrics/sad.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        >>> import numpy as np
+        >>>
+        >>> sad = SAD()
+        >>> prediction = np.zeros((32, 32), dtype=np.uint8)
+        >>> groundtruth = np.ones((32, 32), dtype=np.uint8) * 255
+        >>> sad(prediction, groundtruth)  # doctest: +ELLIPSIS
+        {'SAD': ...}
+    """
+
+    def __init__(self, norm_const=1000, **kwargs) -> None:
```
Comment (LeoXing1996) [inline mmeval/metrics/sad.py:36]:
```suggestion
    def __init__(self, norm_const: int = 1000, **kwargs) -> None:
```

**Comment by LeoXing1996** [comment]
LGTM 🚀

File: tests/test_metrics/test_sad.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+
+from mmeval.metrics import SAD
+
+
+def test_mse():
```
Comment (ice-tong) [inline tests/test_metrics/test_sad.py:7]:
```suggestion
def test_sad():
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**